### PR TITLE
Add support for skipping stage 2 tests

### LINKED ISSUE #5589: Teach Azure Pipelines to skip stage 2 tests when [ci skip] is in the commit message
We need at least stage 1 to pass to merge any PR.

### PR DESCRIPTION
**Pull request by ConorMacBride**
### Description
- Implements CI skipping as documented in [OpenAstronomy/azure-pipelines-templates](https://openastronomy-azure-pipelines.readthedocs.io/en/latest/check_skip.html).
- Also deletes the `ref: master` from the `OpenAstronomy` connection. [Azure will use the default branch anyway.](https://docs.microsoft.com/en-us/azure/devops/pipelines/repos/multi-repo-checkout?view=azure-devops#checking-out-a-specific-ref)

Fixes #5589

### CODE REVIEW

**Comment by ConorMacBride** [comment]
Oh no! The `check-ci.yml` wasn't tested with a fork before. It'll need to be updated upstream to `git clone` the fork instead, if the PR branch is from a fork.

**Comment by ConorMacBride** [comment]
This is now fixed, but need to wait for fix to be merged upstream before merging this PR. (Also need to reset the referenced template branch in this PR.)

**Review by Cadair — APPROVED** [review]

**Comment by Cadair** [comment]
Thanks again!

**Pull request merged**
